awx/ui_next/src/screens/Credential/shared/CredentialPlugins/CredentialPluginPrompt/MetadataStep.jsx

```diff
@@ -0,0 +1,157 @@
+  );
+
+  useEffect(() => {
+    fetchMetadataOptions();
+  }, [fetchMetadataOptions]);
+
+  const testMetadata = () => {
+    // https://github.com/ansible/awx/issues/7126
+  };
+
+  if (isLoading) {
+    return <ContentLoading />;
+  }
+
+  if (error) {
+    return <ContentError error={error} />;
+  }
+
+  return (
+    <>
+      {fields.length > 0 && (
+        <Form>
+          <FormFullWidthLayout>
+            {fields.map(field => {
+              if (field.type === 'string') {
+                if (field.choices) {
+                  return (
+                    <FormGroup
+                      key={field.id}
+                      fieldId={`credential-${field.id}`}
+                      label={field.label}
+                      isRequired={field.required}
+                    >
+                      {field.help_text && (
+                        <Tooltip content={field.help_text} position="right">
+                          <QuestionCircleIcon />
+                        </Tooltip>
+                      )}
+                      <AnsibleSelect
+                        name={`inputs.${field.id}`}
+                        value={form.values.inputs[field.id]}
+                        id={`credential-${field.id}`}
+                        data={field.choices.map(choice => {
+                          return {
+                            value: choice,
+                            key: choice,
+                            label: choice,
+                          };
+                        })}
+                        onChange={(event, value) => {
+                          form.setFieldValue(`inputs.${field.id}`, value);
+                        }}
+                        validate={field.required ? required(null, i18n) : null}
+                      />
+                    </FormGroup>
+                  );
+                }
+
+                return (
+                  <FormField
+                    key={field.id}
+                    id={`credential-${field.id}`}
+                    label={field.label}
+                    tooltip={field.help_text}
+                    name={`inputs.${field.id}`}
+                    type={field.multiline ? 'textarea' : 'text'}
+                    isRequired={field.required}
+                    validate={field.required ? required(null, i18n) : null}
+                  />
+                );
+              }
+
+              return null;
+            })}
+          </FormFullWidthLayout>
+        </Form>
+      )}
+      <Tooltip
+        content={i18n._(
+          t`Click this button to verify connection to the secret management system using the selected credential and specified inputs.`
+        )}
+        position="right"
+      >
+        <TestButton
+          variant="primary"
+          type="submit"
+          onClick={() => testMetadata()}
+        >
+          {i18n._(t`Test`)}
+        </TestButton>
+      </Tooltip>
+    </>
+  );
+}
+
+export default withI18n()(MetadataStep);
```

Comment:
i notice this is not using Formik...is that gonna cause any differences from our other forms (validation, styling, etc.)?

Reply:
Ah yea - this component actually pulls the Formik state using `useFormikContext` here: https://github.com/ansible/awx/pull/7125/files#diff-98c8c894b8812d2914c1799d09dceb4cR26 so I think it should still have the same validation/styling afaik.